accepted/global-json-updates.md

```diff
@@ -0,0 +1,315 @@
+
+Given x.y.znn
+
+* x is the major version
+* y is the minor version
+* z is the feature band
+* nn is the patch
+
+## Current behavior
+
+The SDK resolvers (in both the CLI and VS) search from the current location (project in VS) up the directory hierarchy until they encounter a `global.json` file. When the resolver finds one, _it stop looking_ and attempts to resolve the SDK version specified in that file. This happens even if this `global.json` does not specify an SDK version.
+
+When a `global.json` is encountered and it specifies an SDK version, the SDK resolvers attempt to locate that specific SDK patch version:
+
+1. When the resolver finds no `global.json` or finds one that does not specify SDK version, the latest SDK is used, regardless of whether it is preview or stable.
+2. When the resolver finds a `global.json`, the specific SDK patch version it specifies is used if it is found.
+3. When the resolver finds a `global.json`, and the SDK it specifies can't be found, the resolver looks for a close match:
+    * Prior to .NET Core 2.1
+      * The highest patch version for the SDK feature band is used _regardless of whether it is higher than what is specified_. For example, if `global.json` specifies 2.2.103 is specified and is not available, and 2.2.101 is the highest SDK on the machine, it will be used.
+    * In .NET Core 2.1 and higher
+      * The highest patch version for the SDK feature band is used _assuming it is higher than what is specified_. For example, if `global.json` specifies 2.2.103 is specified, and 2.2.101 is the highest SDK on the machine, an error will occur.
+
+Due to the rules above, the following occurs:
+
+* Due to #2, simply installing a patch on the machine will not update the machine to use the safer version.
+  * The impact of this is mitigated by work we did in .NET Core 3.0 Preview 3 to replace earlier patch versions.
+* Due to #3, users can not guarantee they are pinned to a specific version.
+
+## Proposal
+
+### Roll-forward
+
+.NET Core SDK "roll-forward" is problematic because version numbers are not linear. For example: 2.1.600 contains MSBuild 16 while 2.2.100 contains MSBuild 15.9. This is not a new problem, since it already occurs when there is no `global.json`. However, we do not want to roll-forward in scenarios that might surprise the user. Because of this, roll-forward will be allowed only in limited scenarios.
+
+### Details
+
+The `global.json` schema to offers support for all of the identified scenarios that can be fixed without altering how we find `global.json`:
+
+* `sdk`
+  * `version`
+    * The lowest SDK the resolver would select - any SDKs with lower versions are ignored.
+    * Wildcards are not supported.
+    * Default: empty
+  * `ignorePreview`
+    * `true` or `false`
+    * Default: `false` to match previous behavior.
+    * Alternate names: `releaseOnly` or `useReleaseOnly`
+    * This is new behavior.
+  * `rollForward`
```

Comment:
What would be the impact of declaring `rollForward` but not specifying `sdk.version`? It seems this property is only meaningful if `sdk.version` is specified correct?